x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/bytes.test.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setup, SetupResult } from './processor.helpers';
+
+// Default parameter values automatically added to the Bytes processor when saved
+const defaultBytesParameters = {
+  ignore_failure: undefined,
+  description: undefined,
+};
+
+describe('Processor: Bytes', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { addProcessor, saveNewProcessor, addProcessorType },
+      form,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Click submit button without entering any fields
+    await saveNewProcessor();
+
+    // Expect form error as a processor type is required
+    expect(form.getErrorsMessages()).toEqual(['A type is required.']);
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType({ type: 'bytes', label: 'Bytes' });
```

Comment:
It might make the code a little more readable if we extracted this type object into a variable that can be reused here and elsewhere.

```js
// This could go on line 16
const BYTES_TYPE = { type: 'bytes', label: 'Bytes' };

// Then it can be consumed this way
await addProcessorType(BYTES_TYPE);

```

Reply:
I would approach this slightly differently and change the `addProcessorType` helper to take a single string as argument:

```ts
    async addProcessorType(type: string) {
      await act(async () => {
        find('processorTypeSelector.input').simulate('change', [{ value: type }]);
      });
      component.update();
    },
```

Then when you use it:

```ts
await addProcessorType('bytes'); // if used repeatedly extract `bytes` to its own constant
```

Reply:
Done